### ISSUE TITLE
[Request] sprite size customization

### ISSUE THREAD

**Issue by Level0gamedev** [chrismaltby/gb-studio#10]
As gb works on 8x8 graphics chunks it would be beneficial to allow different sprite sizes (8x8, 8x16, 16x8). It would definitely help with memory concerns for sprite heavy games and allow for more artistic expression.

**Comment by ISSOtm** [comment]
Do you mean hardware sprites? Those can only be 8x8 or 8x16 (taller than wide), that's how the hardware works.

**Comment by sneslink** [comment]
Even if we could only have 8x8 or 16x16: it would still be a nice option to have.

**Comment by HeyItsLollie** [comment]
@ISSOtm I'm assuming they meant ["sprites" as defined by the documentation](https://www.gbstudio.dev/docs/sprites/), where a single "sprite" frame is 16x16.

I would like to see more variance in sprite sizes. I personally don't need nine different actors on screen at once, so I'd be happy to trade graphics memory for taller actors (24x16 or 32x16)... although that said, I also wouldn't need all my sprites to share that same height - only a select few.

If there was some way to determine how much sprite memory is being used in a scene, based on the 8x8 (or 8x16) tiles that have been loaded, that could offer a greater amount of graphical flexibility for users. I imagine it wouldn't be difficult to convey to users that larger sprites = greater memory usage, and to encourage designing around these limits appropriately.

**Comment by LoneDev6** [comment]
Any news on this?

**Comment by Takuyax** [comment]
Having access to the basic 8x8 sprites would be very helpful. I've been experimenting with the 2.0 beta's support for point and click scenes, and tried to use sprites to add some tiny color details to the background, but quickly ran out of sprites.

**Comment by Barthdry** [comment]
I think maybe it will be possible in gba only not in gbc which may allow only 8*8

**Comment by lkcampbell** [comment]
@level0dev or @chrismaltby I believe everything listed in this issue, including things added in via the comments, has been addressed. If so, this issue can be closed as completed:

https://github.com/chrismaltby/gb-studio/blob/develop/CHANGELOG.md#300:
```
Added sprite editor with support for large sprites and multiple animation states
```

https://github.com/chrismaltby/gb-studio/blob/develop/CHANGELOG.md#added-4:
```
Add ability to set Sprite Mode, allowing 8x8 tiles to be used (previously only 8x16 were supported). Sprite Mode can be set globally in Settings and overridden per scene or sprite using the sidebar dropdown. Note that all sprites within a scene must use the same mode as the scene itself.
```

**Comment by chrismaltby** [comment]
Yep this is supported now :-) Thanks for helping tidy up these issues @lkcampbell
